### ISSUE TITLE
Can't find Quick Dial settings location in Firefox

### ISSUE THREAD

**Issue by Ainson** [MatMoul/quickdial-webext#137]
Once again I lost all settings when Quick Dial for some reason reverted to default, but whereas before I could just copy my backup storage.js file back to C:\Users\MyUserName\AppData\Roaming\Mozilla\Firefox\Profiles\MyFirefoxProfileName.default\browser-extension-data\quickdial@matmoul.com in Windows 7 (64), this no longer has any effect.

**Comment by MatMoul** [comment]
Sorry for that but the storage method of FF has changed...
It use now a DB, I'll try to provide an import/export feature but I'm missing time...
Don't miss that Windows have file hostory if you backup your system and MacOS too with timemachine.

**Comment by Ainson** [comment]
Ah that explains it.

An import/export feature would be most welcome since it takes so much time to manually rebuild.

**Comment by paulo-dc** [comment]
I got same issue :( . Lost my settings on one of my forifox profile.

I got many firefox profiles on same PC. I use sync to synchronize bookmarks and other stuff between them. Unfortunatly, not able to synchronize quickdial settings. Even if I copy all content of a profile folder into another one, that does not work.

I wonder where is the db where quickdial settings are ?

**Comment by Dunkh4n** [comment]
I really would want to know that too

Really enjoy your work so far :)

**Comment by MatMoul** [comment]
Look at this folder : [profilename]/storage

For me, this is the path :
[profilename]/storage/default/moz-extension+++51b80cf3-5957-4bd8-8279-b6972ee236cf^userContextId=4294967295/idb/3647222921wleabcEoxlt-eengsairo.files/1011

**Comment by Ainson** [comment]
No such (or similar) folder here

**Comment by MatMoul** [comment]
On Windows, the profile is splitted on appdata/local and appdata/roaming, you will found the data in appdata/roaming.
Don't miss to adapt the ID 51b80cf3-5957-4bd8-8279-b6972ee236cf to your own

**Comment by Ainson** [comment]
Yeah that's where I looked but nothing even if I search for any of the strings in your folder name, so it's either different for each user or Windows version (I have Win 7 (64-bit)).

**Comment by MatMoul** [comment]
@Ainson On Win7 to Win10 Firefox profiles is stored on the same places.
As info AppData is a hidden folder in you Windows Profile :
C:\Users\[UserName]\AppData\
Then : Local\Mozilla and Roaming\Mozilla
When when I reinstall a computer, I just copy these two folders

**Comment by Ainson** [comment]
Yes that was my folder too:

C:\Users\MyUserName\AppData\Roaming\Mozilla\Firefox\Profiles\MyFireFoxProfileName\storage\default

I guess just backing up and restoring that folder should do the trick, but worried when paulo-dc says that doesn't always work..

**Comment by MatMoul** [comment]
Just to be sure, you need to copy 
C:\Users\MyUserName\AppData\Local\Mozilla\Firefox
and
C:\Users\MyUserName\AppData\Roaming\Mozilla\Firefox

An other idea is that after a first launch of FF you use a new default profile

**Comment by Ainson** [comment]
Gotcha.

I made a .bat file to copy those two folders, so should be back to normal.

Should be said that no QuickDial settings lost with the new db system. With the old I would probably have had an episode already as it was not infrequent.

**Comment by paulo-dc** [comment]
hello, thank you for the info.

But When I copy the file in the folder ../3647222921wleabcEoxlt-eengsairo.files/xxx from a profile to another, only url are "synced" . Not all parameters like images.

And when I copy also the *.sqlite db, the quickdial page becomes blank on the destination profile

The only way I succeed was by copying the entire profile into another one.

**Comment by Ainson** [comment]
Well today all settings were gone, and even though I had a recent complete backup of these folders:

C:\Users\MyUserName\AppData\Local\Mozilla\Firefox
and
C:\Users\MyUserName\AppData\Roaming\Mozilla\Firefox

.. it didn't help at all, so I had to do all dials manually (putting a picture on each, and re-organizing them) which was a pain when I have almost 2 pages..

Any idea why both profiles would be corrupted? Seems too strange to be real. It's like Firefox changed the fundamentals again. (ver 81.0.1 (64 bit))
